### PR TITLE
Integration and User password reset

### PR DESCRIPTION
**Pull request by lpatte**
- Integration (resource/datasource/import/doc)
- Add Password Reset system to user resources (resource/doc)
- Generalize Certificate resources (datasource/doc)

### CODE REVIEW

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 1]
```diff
@@ -40,6 +40,8 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Name of the user.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update.
+
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:44]:
```
output "user_password" {
    value = ovh_cloud_project_database_m3db_user.user.password
    sensitive = true
}
```

Comment (scraly) [reply website/docs/r/cloud_project_database_m3db_user.html.markdown:44]: speak about the `terraform refresh` command to execute after the apply ... :)

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 2]
```diff
@@ -40,6 +40,8 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Name of the user.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update.
```
[the remark this answers is not in the document]
Comment (lpatte) [reply website/docs/r/cloud_project_database_m3db_user.html.markdown]: no it's for all engines

File: website/docs/r/cloud_project_database_integration.html.markdown [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  Creates an integration for a database cluster associated with a public cloud project.
+---
+
+# ovh_cloud_project_database_integration
+
+Creates an integration for a database cluster associated with a public cloud project.
+
+With this resource you can create an integration for all engine exept `mongodb`.
+
+Thanks to the integration feature you can have your postgreSQL logs in your OpenSearch DB
```
Comment (scraly) [inline website/docs/r/cloud_project_database_integration.html.markdown:15]:
```suggestion
Please take a look at the list of available `types` in the Àrgument references` section in order to know the list of avaulable integrations. For example, thanks to the integration feature you can have your PostgreSQL logs in your OpenSearch Database.
```

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 4]
```diff
@@ -25,6 +27,35 @@ resource "ovh_cloud_project_database_m3db_user" "user" {
...
   name          = "johndoe"
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_m3db_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:37]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 5]
```diff
@@ -25,6 +27,35 @@ resource "ovh_cloud_project_database_m3db_user" "user" {
...
+Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:38]:
```suggestion
Use command `terraform refresh` after executing `terraform apply` to update the output with new password.
```

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 6]
```diff
@@ -40,6 +71,8 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Name of the user.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:74]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after executing `terraform apply` to update the output with new password.
```

File: website/docs/r/cloud_project_database_mongodb_user.html.markdown [hunk 7]
```diff
@@ -12,6 +12,8 @@ Creates an user for a mongodb cluster associated with a public cloud project.
 
 ## Example Usage
 
+Create a user johndoe in a mongodb database.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_mongodb_user.html.markdown:15]:
```suggestion
Create a user johndoe in a MongoDB database.
```

File: website/docs/r/cloud_project_database_mongodb_user.html.markdown [hunk 8]
```diff
@@ -25,6 +27,34 @@ resource "ovh_cloud_project_database_mongodb_user" "user" {
...
   roles         = ["backup", "readAnyDatabase"]
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_mongodb_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_mongodb_user.html.markdown:37]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_mongodb_user.html.markdown [hunk 9]
```diff
@@ -25,6 +27,34 @@ resource "ovh_cloud_project_database_mongodb_user" "user" {
...
+Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_mongodb_user.html.markdown:38]:
```suggestion
Use command `terraform refresh` after executing `terraform apply` to update the output with new password.
```

File: website/docs/r/cloud_project_database_mongodb_user.html.markdown [hunk 10]
```diff
@@ -38,14 +68,17 @@ The following arguments are supported:
...
+* `roles` - (Optional: if omit, default role) Roles the user belongs to.
+Available roles:
   * `backup`
   * `dbAdminAnyDatabase`
   * `readAnyDatabase`
   * `readWriteAnyDatabase`
   * `restore`
   * `userAdminAnyDatabase`
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_mongodb_user.html.markdown:80]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use the `terraform refresh` command after `terraform apply` to update the output with the new password.
```

File: website/docs/r/cloud_project_database_opensearch_user.html.markdown [hunk 11]
```diff
@@ -12,6 +12,8 @@ Creates an user for a opensearch cluster associated with a public cloud project.
 
 ## Example Usage
 
+Create a user johndoe in a opensearch database.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_opensearch_user.html.markdown:15]:
```suggestion
Create a user johndoe in a OpenSearch database.
```

File: website/docs/r/cloud_project_database_opensearch_user.html.markdown [hunk 12]
```diff
@@ -54,6 +92,8 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Username affected by this acl.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_opensearch_user.html.markdown:95]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use the `terraform refresh` command after `terraform apply` to update the output with the new password.
```

File: website/docs/r/cloud_project_database_postgresql_user.html.markdown [hunk 13]
```diff
@@ -12,6 +12,8 @@ Creates an user for a postgresql cluster associated with a public cloud project.
 
 ## Example Usage
 
+Create a user johndoe in a postgresql database.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_postgresql_user.html.markdown:15]:
```suggestion
Create a user johndoe in a PostgreSQL database.
```

File: website/docs/r/cloud_project_database_opensearch_user.html.markdown [hunk 14]
```diff
@@ -32,6 +34,41 @@ resource "ovh_cloud_project_database_opensearch_user" "user" {
...
   name          = "johndoe"
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_opensearch_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_opensearch_user.html.markdown:44]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_postgresql_user.html.markdown [hunk 15]
```diff
@@ -25,6 +27,34 @@ resource "ovh_cloud_project_database_postgresql_user" "user" {
...
   roles         = ["replication"]
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_postgresql_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_postgresql_user.html.markdown:37]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_postgresql_user.html.markdown [hunk 16]
```diff
@@ -38,8 +68,11 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Name of the user.
 
-* `roles` - (Optional: if omit, default role) Roles the user belongs to. Possible values:
-  * `["replication"]`
+* `roles` - (Optional: if omit, default role) Roles the user belongs to.
+  Available roles:
+  * `replication`
+
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_postgresql_user.html.markdown:75]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use the `terraform refresh` command after `terraform apply` to update the output with the new password.

```

File: website/docs/r/cloud_project_database_redis_user.html.markdown [hunk 17]
```diff
@@ -12,6 +12,8 @@ Creates an user for a redis cluster associated with a public cloud project.
 
 ## Example Usage
 
+Create a user johndoe in a redis database.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_redis_user.html.markdown:15]:
```suggestion
Create a user johndoe in a Redis database.
```

File: website/docs/r/cloud_project_database_redis_user.html.markdown [hunk 18]
```diff
@@ -28,6 +30,37 @@ resource "ovh_cloud_project_database_redis_user" "user" {
...
   name          = "johndoe"
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_redis_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_redis_user.html.markdown:40]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_redis_user.html.markdown [hunk 19]
```diff
@@ -28,6 +30,37 @@ resource "ovh_cloud_project_database_redis_user" "user" {
...
+  engine        = "redis"
+  id            = "ZZZZ"
+}
+
+resource "ovh_cloud_project_database_redis_user" "user" {
+  service_name    = data.ovh_cloud_project_database.redis.service_name
+  cluster_id      = data.ovh_cloud_project_database.redis.id
+  categories      = ["+@set", "+@sortedset"]
+  channels        = ["*"]
+  commands	      = ["+get", "-set"]
```
Comment (scraly) [inline website/docs/r/cloud_project_database_redis_user.html.markdown:54]:
```suggestion
  commands	  = ["+get", "-set"]
```

File: website/docs/r/cloud_project_database_redis_user.html.markdown [hunk 20]
```diff
@@ -28,6 +30,37 @@ resource "ovh_cloud_project_database_redis_user" "user" {
...
+  keys		        = ["data", "properties"]
```
Comment (scraly) [inline website/docs/r/cloud_project_database_redis_user.html.markdown:55]:
```suggestion
  keys		  = ["data", "properties"]
```

File: website/docs/r/cloud_project_database_redis_user.html.markdown [hunk 21]
```diff
@@ -49,6 +82,8 @@ The following arguments are supported:
 
 * `name` - (Required, Forces new resource) Name of the user.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_redis_user.html.markdown:85]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use the `terraform refresh` command after `terraform apply` to update the output with the new password.

```

File: website/docs/r/cloud_project_database_user.html.markdown [hunk 22]
```diff
@@ -32,6 +34,34 @@ resource "ovh_cloud_project_database_user" "user" {
...
   name          = "johndoe"
 }
+
+output "user_password" {
+    value = ovh_cloud_project_database_user.user.password
+    sensitive = true
+}
+```
+
+To reset password of the user previouly create, update the `password_reset` attribut.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_user.html.markdown:44]:
```suggestion
-> __NOTE__ To reset password of the user previously created, update the `password_reset` attribute.
```

File: website/docs/r/cloud_project_database_user.html.markdown [hunk 23]
```diff
@@ -52,17 +82,21 @@ Available engines for this resource (other have specific resource):
 
 * `name` - (Required, Forces new resource) Name of the user.
 
+* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use command `terraform refresh` after `terraform apply` to update output with new password.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_user.html.markdown:85]:
```suggestion
* `password_reset` - (Optional) Arbitrary string to change to trigger a password update. Use the `terraform refresh` command after `terraform apply` to update the output with the new password.

```

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — APPROVED** [review]

**Pull request merged**